### PR TITLE
Add Duration overloads to set Java timeouts

### PR DESCRIPTION
**Pull request by pepone**
A follow up to #2794 but for Java

### CODE REVIEW

File: java/src/Ice/src/main/java/com/zeroc/Ice/ObjectPrx.java [hunk 1]
```diff
@@ -290,14 +292,14 @@ java.util.concurrent.CompletableFuture<com.zeroc.Ice.Object.Ice_invokeResult> ic
      * @return The locator cache timeout value (in seconds).
      * @see Locator
      */
-    int ice_getLocatorCacheTimeout();
+    Duration ice_getLocatorCacheTimeout();
```
Comment (bernardnormier) [inline java/src/Ice/src/main/java/com/zeroc/Ice/ObjectPrx.java:295]: The doc-comment is not correct. The returned Duration is _not "in seconds".

File: java/src/Ice/src/main/java/com/zeroc/Ice/ObjectPrx.java [hunk 2]
```diff
@@ -290,14 +292,14 @@ java.util.concurrent.CompletableFuture<com.zeroc.Ice.Object.Ice_invokeResult> ic
...
 
     /**
      * Returns the invocation timeout of this proxy.
      *
      * @return The invocation timeout value (in milliseconds).
      */
-    int ice_getInvocationTimeout();
+    Duration ice_getInvocationTimeout();
```
Comment (bernardnormier) [inline java/src/Ice/src/main/java/com/zeroc/Ice/ObjectPrx.java:302]: Same here: please fix the doc-comment.

File: java/src/Ice/src/main/java/com/zeroc/Ice/ObjectPrx.java [hunk 3]
```diff
@@ -331,6 +333,15 @@ java.util.concurrent.CompletableFuture<com.zeroc.Ice.Object.Ice_invokeResult> ic
...
+     * Returns a proxy that is identical to this proxy, except for the locator cache timeout.
+     *
+     * @param newTimeout The new locator cache timeout (in seconds).
+     * @return The proxy with the new timeout.
+     * @see Locator
+     */
+    ObjectPrx ice_locatorCacheTimeout(Duration newTimeout);
+
     /**
      * Returns a proxy that is identical to this proxy, except for the invocation timeout.
```
Comment (bernardnormier) [inline java/src/Ice/src/main/java/com/zeroc/Ice/ObjectPrx.java:346]: Here we need to specify the unit

File: java/src/Ice/src/main/java/com/zeroc/Ice/ObjectPrx.java [hunk 4]
```diff
@@ -339,6 +350,14 @@ java.util.concurrent.CompletableFuture<com.zeroc.Ice.Object.Ice_invokeResult> ic
      */
     ObjectPrx ice_invocationTimeout(int newTimeout);
 
+    /**
+     * Returns a proxy that is identical to this proxy, except for the invocation timeout.
+     *
+     * @param newTimeout The new invocation timeout (in milliseconds).
```
Comment (bernardnormier) [inline java/src/Ice/src/main/java/com/zeroc/Ice/ObjectPrx.java:356]: Fix doc-comment.

**Review by bernardnormier — APPROVED** [review]
It would be nice to fix the C++ doc-comments from the previous PR too.

**Review by externl — APPROVED** [review]

File: java/src/Ice/src/main/java/com/zeroc/Ice/ObjectPrx.java (hunk 3 above)
Comment (pepone) [reply java/src/Ice/src/main/java/com/zeroc/Ice/ObjectPrx.java:346]: We do in the param doc `The new invocation timeout (in milliseconds).`

**Pull request merged**